week-14/homework/index.js

```diff
@@ -0,0 +1,157 @@
+        if (!title) {
+            title = data[i].title;
+        }
+        list.innerHTML += `
+        <div class="col-6 col-md-3 mb-5 text-center" title="${title}">
+          <a href="details.html?id=${data[i].mal_id}" class="link-primary d-block position-relative text-decoration-none" data-component="item">
+            <span class="position-absolute badge bg-danger top-0 end-0">
+                <i class="bi bi-star-fill"></i> ${data[i].score}
+            </span>
+            <span class="d-flex flex-column justify-content-center">
+                <img class="rounded shadow" src="${data[i].images.jpg.image_url}" data-component="image">
+                <span class="text-dark mt-2" data-component="title">${title}</span>
+            </span>
+          </a>
+        </div>`;
+    }
+
+    //renders the pagination selection
+    let page = animeData.pagination;
+    currentPage = page.current_page;
+    totalPages = page.items.count;
+
+    //clears page
+    selectPagination.innerHTML = '';
+    //loop through pages and generate html
+    for (i = 1; i <= totalPages; i++) {
+        if (i == currentPage) {
+            selectPagination.innerHTML += `<option selected value="${i}">${i}</option>`;
+        } else {
+            selectPagination.innerHTML += `<option value="${i}">${i}</option>`;
+        }
+    }
+
+    //hide loader and re-enable buttons when data received
+    loader.classList.add('d-none');
+    selectPagination.disabled = false;
+    nextBtn.disabled = false;
+    prevBtn.disabled = false;
+    filter.disabled = false;
+}
+
+let renderFilter = function (genresData) {
+    let allGenres = genresData.data;
+
+    //render filter dropdown
+    filter.innerHTML = '';
+    if (currentFilter == 'top') {
+        filter.innerHTML += `<option selected value="top">Top anime</option>`;
+    } else {
+        filter.innerHTML += `<option value="top">Top anime</option>`;
+    }
+    for (let i = 0; i < allGenres.length; i++) {
+        if (currentFilter == allGenres[i].mal_id) {
+            filter.innerHTML += `<option selected value="${allGenres[i].mal_id}">${allGenres[i].name} (${allGenres[i].count})</option>`;
+        } else {
+            filter.innerHTML += `<option value="${allGenres[i].mal_id}">${allGenres[i].name} (${allGenres[i].count})</option>`;
+        }
+    }
+}
+
+//fetch from api
+let fetchData = function () {
+    //remove d-none to show loader and disable buttons while loading
+    loader.classList.remove('d-none');
+    selectPagination.disabled = true;
+    nextBtn.disabled = true;
+    prevBtn.disabled = true;
+    filter.disabled = true;
+
+    //add loading animation
+    let allListItems = list.querySelectorAll('div');
+    for (let i = 0; i < allListItems.length; i++) {
+        allListItems[i].classList.add('animate-pulse');
+    }
+
+    //Checks if current filter is 'top' else renders genres
+    if (currentFilter == 'top') {
+        fetch(`https://api.jikan.moe/v4/top/anime?page=${currentPage}`).
```

Comment:
Do not use `==`, in Javascript, we should always use strictly equal `===`

Also, the code in here is duplicated with the code in line 103 and could be simplified by 
```
let requestUrl = 'https://api.jikan.moe/v4';

if (currentFilter === 'top') {
   requestUrl = `${requestUrl}/top/anime?page=${currentPage}`;
else {
   requestUrl = `${requestUrl}/top/anime?genres=${currentFilter}&page=${currentPage}`;
}

fetch(requestUrl)
  .then(function (response) {
       return response.json();
   })
   .then(renderData);
```